tests/memory/enc/basic.c

```diff
@@ -102,11 +102,6 @@ void test_realloc(void)
 #pragma GCC diagnostic pop
 #endif
     OE_TEST(ptr2 == NULL);
-
-    /* realloc(X, 0) is implementation defined, but we can always free it. */
-    free(ptr);
-    ptr = realloc(NULL, 0);
-    free(ptr);
```

Comment:
Why was this removed?

Reply:
Reverted it. There is still a problem when attempting to release already-freed ptr at line 106 (assume ptr != NULL and ptr is released at line 100), but I will try to deal with it after other warnings are settled.

Reply:
I thought about it for a while and reverted to the original version.

So there are 3 scan-build warnings, one for double-free and the others for memory leaks. Since the test is intended to ensure that memory allocation fails at the end of each unit test, it is not likely to change OE_TEST(ptr == NULL) and OE_TEST(ptr2 == NULL).

The double-free warning occurs because scan-build does not recognize oe_abort(), or it thinks that abort() is different than exit(). Because if ptr is really released at line 100, it means that ptr2 is successfully reallocated, therefore failing the test and leaving the rest of the code unreachable.